### PR TITLE
Re-Organized includes

### PR DESCRIPTION
**Pull request by slizzered**
- Files now more often directly include what they need
 - some unnecesssary includes removed
 - removed custom thrust_nowarn headers, since warning no longer happens
 - some of the removed includes did conflict with windows, so this also helps a little with #81

I did use eclipse to get help on what to change. Since eclipse is not perfect in doing so, I checked everything manually. I might have missed something. However, it builds without warnings on my machine.

### CODE REVIEW

**Comment by slizzered** [comment]
Something I just thought of.... 
Old binary size: **4.9 MB**
New binary size after this PR is merged: **4.3 MB** :dancers:

File: src/for_loops_clad.cc [hunk 1]
```diff
@@ -18,8 +18,8 @@
  * If not, see <http://www.gnu.org/licenses/>.
  */
 
-#include <math.h>
-#include <stdlib.h>
+#include <cmath>
+#include <cstdlib>
 #include <vector>
 #include <stdio.h>
```
Comment (psychocoderHPC) [inline src/for_loops_clad.cc]: this should be `cstdio` in C++

Comment (slizzered) [inline src/for_loops_clad.cc]: Thanks! I missed that one opportunity for replacement!

**Comment by erikzenker** [comment]
Ready to merge :sparkles:

**Pull request merged**